src/sentry/workflow_engine/endpoints/validators/base/detector.py

```diff
@@ -63,6 +67,22 @@ def update(self, instance: Detector, validated_data: dict[str, Any]):
         instance.name = validated_data.get("name", instance.name)
         instance.type = validated_data.get("detector_type", instance.group_type).slug
 
+        # Handle enable/disable detector
+        if "enabled" in validated_data:
+            enabled = validated_data.get("enabled")
+            # get data source and check if the type is metric
+            if DataSource.objects.filter(
+                detectors__in=[instance], type=DATA_SOURCE_SNUBA_QUERY_SUBSCRIPTION
+            ).exists():
+                update_detector(instance, enabled)
+            else:
+                # TODO this is dupe code from update_detector, find a shared location
+                updated_detector_status = ObjectStatus.ACTIVE if enabled else ObjectStatus.DISABLED
```

Comment:
🤔 - i feel like the snuba query code should live in a specific validator for that data source type. instead of this having an if statement to filter the types, can we have a hook into the specific data sources?

let me know if you have any questions on this or want to pair on this!

Reply:
👍 I moved disabling the query subscriptions to the metric issue detector validator's update method, and the base class does the detector level disabling